src/main/scala/com/fulcrumgenomics/umi/CorrectUmis.scala

```diff
@@ -77,8 +83,8 @@ object CorrectUmis {
     */
   def findUmiPairsWithinDistance(umis: Seq[String], distance: Int): Seq[(String,String,Int)] = {
     umis.tails.flatMap {
-      case x +: ys => ys.map(y => (x, y, Sequences.countMismatches(x, y))).filter(_._3 <= distance)
-      case _      => Seq.empty
+      case x +: ys => ys.map(y => (x, y, Sequences.countMismatchesWithMax(x.getBytes, y.getBytes, distance+1))).filter(_._3 <= distance)
+      case _       => Seq.empty
     }.toList
   }
```

Comment:
Looking at this code anew, I wonder why we chose to do `umi.tails.flatMap`.  I think this would likely be considerably more efficient as something like:

```suggestion
val builder = Seq[(String, String,Int)].builder()
forloop (from=0, until=umis.length) { i =>
  forloop (from=i+1, until=umis.length) { j => 
    val d = Sequences.countMismatchesWithMax(umis(i).getBytes, umis(j).getBytes, distance+1)
    if (d <= distance) builder += (umis(i), umis(j), d)
  }
}

builder.result()
```